framework/CodeInterfaces/PHISICS/phisicsdata.py

```diff
@@ -0,0 +1,961 @@
+      @ In, matchedTimeSteps, list, list of time steps considered
+      @ Out, decayLabelList, list, list of the decay labels (under the format fuel|isotope)
+      @ Out, decayList, list, list of the decay heat values
+    """
+    materialList = []
+    decayLabelList = []
+    decayList = []
+    timeStepList = []
+    for mpi in xrange(0,len(self.instantOutputFileMPI)):
+      decayFlag = 0 
+      breakFlag = 0 
+      matLineFlag = 0 
+      materialCounter = 0
+      with open(os.path.join(self.workingDir, self.instantOutputFileMPI[mpi])) as outfile:
+        for line in outfile:
+          if re.search(r'Decay Heat computed',line) and mpi == 0:
+            self.isDecayHeatPrinted, self.decayHeatUnits = self.findDecayHeat(line)
+            if self.isDecayHeatPrinted is False:
+              decayLabelList = ['decayHeat']
+              decayList = 0
+              return  decayLabelList,decayList
+          if re.search(r'INDIVIDUAL DECAY HEAT BLOCK',line):
+            decayFlag = 1
+            materialCounter = materialCounter + 1
+          if re.search(r'CUMULATIVE DECAY HEAT BLOCK',line):
+            decayFlag = 0
+          if re.search(r'BURNUP OUTPUT',line):
+            breakFlag = 1
+          if decayFlag == 1 and breakFlag == 0:
+            line = re.sub(r'\n',r'',line)
+            decayLine = filter(None, line.split(' '))
+            if decayLine != []:
+              stringIsFloatNumber = self.isFloatNumber(decayLine)
+            if stringIsFloatNumber is True and decayLine != []:
+              if (float(decayLine[0]) == float(matchedTimeSteps[timeStepIndex])):
+                for i in xrange (1,len(self.isotopeList)):
+                  decayLabelList.append('decay'+'|'+self.materialsDict['MPI-'+str(mpi)][materialCounter]+'|'+self.isotopeList[i])
+                  decayList.append(decayLine[i])
+          if breakFlag == 1:
+            break
+    return decayLabelList, decayList
+      
+  def getDecayHeatMrtau(self,timeStepIndex,matchedTimeSteps):
+    """
+      Gets the decay heat from the standalone MRTAU output.
+      @ In, timeStepIndex, integer, timestep number
+      @ In, matchedTimeSteps, list, list of time steps considered
+      @ Out, decayListMrtau, list, list of the decay values from MRTAU
+    """
+    decayFlag = 0
+    breakFlag = 0
+    self.decayLabelListMrtau = []
+    self.numDensityLabelListMrtau = []
+    decayListMrtau = []
+    with open(os.path.join(self.workingDir,self.mrtauCSVOutputMPI), 'r') as outfile:
+      for line in outfile:
+        if re.search(r'INDIVIDUAL DECAY HEAT BLOCK',line):
+          decayFlag = 1
+        if re.search(r'CUMULATIVE DECAY HEAT BLOCK',line):
+          breakFlag = 1
+        if decayFlag == 1 and breakFlag == 0:
+          line = re.sub(r'\n',r'',line)
+          decayLine = filter(None, line.split(' '))
+          if decayLine != []: stringIsFloatNumber = self.isFloatNumber(decayLine)
```

Comment:
isinstance

Reply:
I can't: 
the float is under s tring format. So If I do `isinstance(line[0],float)`, it will return false all the time. 
if I do` isinstance(float(line[0]),float)` it'll error out if the element is not a float. Hence the method